[Bug] Build fails with Protobuf 30 from type changes

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/pulsar-client-cpp/issues) and found nothing similar.


### Version

OS: macOS 15
Client: v3.7.0

### Minimal reproduce step

Should be reproducible with any build attempt using Protobuf 30.0 (C++ 6.30.0) or later, at least when Abseil was built to use `std::string_view`. 

However, won't be able to use Vcpkg build yet as they only have Protobuf 29.3 (C++ 5.29.3).

### What did you expect to see?

Successful build

### What did you see instead?

```
lib/ProtobufNativeSchema.cc:42:23: error: no viable conversion from 'internal::DescriptorStringView' (aka 'basic_string_view<char>') to 'const std::string' (aka 'const basic_string<char>')
   42 |     const std::string rootMessageTypeName = descriptor->full_name();
      |                       ^                     ~~~~~~~~~~~~~~~~~~~~~~~
```

### Anything else?

Cause by breaking change in https://github.com/protocolbuffers/protobuf/commit/d1990d968a54176eb9f4229abe7f7c97ece50cec

Should be possible to use explicit constructor for `std::string` instead and work for both Protobuf 30 and older compatibility. I think move constructor/assignment should avoid unnecessary copies for older Protobuf.

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

--- a/lib/ProtobufNativeSchema.cc
+++ b/lib/ProtobufNativeSchema.cc
@@ -39,8 +39,8 @@ SchemaInfo createProtobufNativeSchema(const google::protobuf::Descriptor* descri
     }
 
     const auto fileDescriptor = descriptor->file();
-    const std::string& rootMessageTypeName = descriptor->full_name();
-    const std::string& rootFileDescriptorName = fileDescriptor->name();
+    const std::string& rootMessageTypeName = std::string(descriptor->full_name());
+    const std::string& rootFileDescriptorName = std::string(fileDescriptor->name());
 
     FileDescriptorSet fileDescriptorSet;
     internalCollectFileDescriptors(fileDescriptor, fileDescriptorSet);

